bindings/java/src/org/sleuthkit/datamodel/SleuthkitCase.java

```diff
@@ -13228,7 +13228,7 @@ PreparedStatement getPreparedStatement(PREPARED_STATEMENT statementKey, int gene
 		PreparedStatement getPreparedStatement(String sqlStatement, int generateKeys) throws SQLException {
 			PreparedStatement statement;
 			String statementKey = "SQL:" + sqlStatement + " Key:" + generateKeys;
-			if (adHocPreparedStatements.containsKey(statementKey)) {
+			if (adHocPreparedStatements.containsKey(statementKey) && !adHocPreparedStatements.get(statementKey).isClosed()) {
```

Comment:
What? The cache is created for each CaseDbConnection - there's no reason to manually empty it when closing the connection. And the statements really shouldn't be closed manually before that point, but I added a check just in case.

Reply:
ok. i misunderstood the cache layer then. I assumed they are not cleared when the collection is closed and returned to pool.

Reply:
Nope - I wanted to, but since each is associated with a specific connection you can't keep using them after the connection is closed.